apps/roam/src/utils/renderNodeTagPopup.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useState, useRef, useEffect } from "react";
+import ReactDOM from "react-dom";
+import { Button, Popover, Position } from "@blueprintjs/core";
+
+let currentPopup: HTMLDivElement | null = null;
+
+export const removeNodeTagPopup = () => {
+  if (currentPopup) {
+    ReactDOM.unmountComponentAtNode(currentPopup);
+    currentPopup.remove();
+    currentPopup = null;
+  }
+};
+
+const HoverControlledPopover = ({
+  onClick,
+  label,
+  onClose,
+  tagElement,
+}: {
+  onClick: () => void;
+  label: string;
+  onClose: () => void;
+  tagElement: HTMLElement;
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const timeoutRef = useRef<number>();
+
+  useEffect(() => {
+    const handleMouseEnter = () => {
+      clearTimeout(timeoutRef.current);
+      setIsOpen(true);
+    };
+
+    const handleMouseLeave = (e: MouseEvent) => {
+      const relatedTarget = e.relatedTarget as HTMLElement;
+      if (relatedTarget?.closest(".discourse-node-tag-portal")) {
+        return;
+      }
+
+      timeoutRef.current = window.setTimeout(() => {
+        setIsOpen(false);
+        setTimeout(onClose, 100);
+      }, 150);
+    };
+
+    tagElement.addEventListener("mouseenter", handleMouseEnter);
+    tagElement.addEventListener("mouseleave", handleMouseLeave);
+
+    return () => {
+      tagElement.removeEventListener("mouseenter", handleMouseEnter);
+      tagElement.removeEventListener("mouseleave", handleMouseLeave);
+      clearTimeout(timeoutRef.current);
+    };
+  }, [tagElement, onClose]);
+
+  const handlePopoverMouseEnter = () => {
+    clearTimeout(timeoutRef.current);
+  };
+
+  const handlePopoverMouseLeave = () => {
+    timeoutRef.current = window.setTimeout(() => {
+      setIsOpen(false);
+      setTimeout(onClose, 100);
+    }, 150);
+  };
+
+  const rect = tagElement.getBoundingClientRect();
+
+  return (
+    <Popover
+      content={
+        <div
+          onMouseEnter={handlePopoverMouseEnter}
+          onMouseLeave={handlePopoverMouseLeave}
+        >
+          <Button intent="primary" outlined onClick={onClick} text={label} />
+        </div>
+      }
+      target={
+        <div
+          style={{
+            position: "fixed",
```

Comment:
We shouldn't need to set the position like this.  It is not centered.
<img width="285" height="119" alt="image" src="https://github.com/user-attachments/assets/0712a3de-a46d-415b-a560-d611ebbcb277" />

Again, if you see our last meeting, the overlay was perfectly centered automagically.

<img width="60" height="54" alt="image" src="https://github.com/user-attachments/assets/78ec3e9c-8a04-4efd-a5b2-e8727a56e1ed" />